environs/bootstrap/bootstrap_test.go

```diff
@@ -128,8 +128,11 @@ func (s *bootstrapSuite) TestBootstrapEmptyConstraints(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(env.bootstrapCount, gc.Equals, 1)
 	env.args.AvailableTools = nil
+	// Should have host arch, since none was explicitly specified.
```

Comment:
if none is specified we do need to use the host arch when looking for tools but i'm not sure of the correctness in setting the constraint attribute - constraints are what the user asked for explicitly